### DISCUSSION TITLE
Translate html/xml report into twig templates

### DISCUSSION THREAD

**Discussion by sebastianneubert** [Codeception/Codeception#6325] [Ideas]
## what 

In case you want to customize the html report template, it would be a good approach to use symfony/twig as template engine. the current template for the html report is a mess and should be refactored anyways.

## how 

Maybe it's a good idea to have something like a general extension which is by default enabled (like the failed extension), generates the reports from the hooks and have different possibilities of outputs (json/xml/html/...)

### requirements:

- new dependency to symfony/twig

**Comment by Naktibalda** [comment]
We actually use template engine, it is very basic template engine used by PHPUnit - https://github.com/sebastianbergmann/php-text-template.
Changing template engine to twig wouldn't improve the look and the quality of HTML.

* If you want to improve the look of standard HTML report, you are welcome to make a pull request.
* If you want to use your own customized templates, the only change necessary is to make template path configurable.

> Maybe it's a good idea to have something like a general extension which is by default enabled (like the failed extension), generates the reports from the hooks and have different possibilities of outputs (json/xml/html/...)

In Codeception 5 reporters will work in a very similar manner to extensions, it is possible to use custom extensions for generating reports in older versions of Codeception too.

Links to relevant resources:
* HTML Reporter used by Codeception 2-4 - https://github.com/Codeception/phpunit-wrapper/blob/9.0/src/ResultPrinter/HTML.php
* Templates used for generating report - https://github.com/Codeception/phpunit-wrapper/tree/9.0/src/ResultPrinter/template
* HTML reporter for Codeception 5 - https://github.com/Codeception/Codeception/blob/5.0/src/Codeception/Reporter/HtmlReporter.php
* New location of templates https://github.com/Codeception/Codeception/tree/5.0/src/Codeception/Reporter/template
